lang/test/org/partiql/lang/eval/EvaluatingCompilerExceptionsTest.kt

```diff
@@ -242,4 +242,37 @@ class EvaluatingCompilerExceptionsTest : EvaluatorTestBase() {
                 Property.COLUMN_NUMBER to 11L,
                 Property.BINDING_NAME to "leading")
         )
+
+    @Test
+    fun intAsNonTextStructField() =
+        checkInputThrowingEvaluationException(
+            "{'apple' : 1, `banana` : 2, 1 : 3}",
+            ErrorCode.EVALUATOR_NON_TEXT_STRUCT_FIELD,
+            mapOf(
+                Property.LINE_NUMBER to 1L,
+                Property.COLUMN_NUMBER to 29L
+            )
+        )
+
+    @Test
+    fun timestampAsNonTextStructField() =
+        checkInputThrowingEvaluationException(
+            "{`2007-02-23T12:14Z` : 3}",
+            ErrorCode.EVALUATOR_NON_TEXT_STRUCT_FIELD,
+            mapOf(
+                Property.LINE_NUMBER to 1L,
+                Property.COLUMN_NUMBER to 2L
+            )
+        )
+
```

Comment:
May want to also add some tests with `NULL` and `MISSING` as the struct field name.